src/Tools/VMC/EvtLibPXSec.h

```diff
@@ -0,0 +1,40 @@
+#ifndef _LLEWELLYN_SMITH_QELCC_CROSS_SECTION_H_
+#define _LLEWELLYN_SMITH_QELCC_CROSS_SECTION_H_
+
+#include "Framework/EventGen/XSecAlgorithmI.h"
+
+#include "Tools/VMC/Key.h"
+
+class TGraph;
+
+namespace genie {
+namespace vmc {
+
+class EvtLibPXSec : public XSecAlgorithmI {
+
+public:
+  EvtLibPXSec();
+  EvtLibPXSec(string config);
+  virtual ~EvtLibPXSec();
+
+  // XSecAlgorithmI interface implementation
+  double XSec            (const Interaction * i, KinePhaseSpace_t k) const;
+  double Integral        (const Interaction * i) const;
+  bool   ValidProcess    (const Interaction * i) const;
+
+  // Override the Algorithm::Configure methods to load configuration
+  // data to private data members
+  void Configure (const Registry & config);
+  void Configure (string param_set);
+
+protected:
+  TGraph* GetXSec(const Interaction* in) const;
+  void LoadXSecs() const;
+
+  mutable std::map<Key, TGraph*> fXSecs;
```

Comment:
why is this mutable?

Reply:
Because Integral() has to be declared const, and I call LoadXSecs() lazily there. I suppose in principle one could call it at the end of Configure()? I don't know what the normal coding convention is. My claim is that the current structure is fine - logically the function is const, it's just an implementation detail that it happens to populate a cache.

Reply:
I worked around this one.